onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -1056,7 +1056,52 @@ CUDAExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph,
       result.push_back(std::make_unique<ComputeCapability>(std::move(sub_graph)));
     }
   }
+
+  if (enable_compiler_) {
+#ifdef USE_TENSORRT
+    auto trt_result = GetTensorrtCapability(graph);
+    if (!trt_result.empty() && !result.empty()) {
+      for (int i = result.size() - 1; i >= 0; i--) {
+        bool in_trt = false;
+        for (auto& capability : trt_result) {
+          for (auto& node : capability->sub_graph->nodes) {
+            if (node == result[i]->sub_graph->nodes[0]) {
+              in_trt = true;
+              break;
+            }
+          }
+        }
+        if (in_trt) {
+          result.erase(result.begin() + i);
+        }
+      }
+      for (auto& c : trt_result) {
+        result.push_back(std::move(c));
+      }
+    }
+#endif
+  }
+
   return result;
 }
 
+common::Status CUDAExecutionProvider::Compile(const std::vector<onnxruntime::Node*>& fused_nodes,
+                                              std::vector<NodeComputeInfo>& node_compute_funcs) {
+  if (enable_compiler_) {
+#ifdef USE_TENSORRT
```

Comment:
Maybe this entire function and its definition should be under #ifdef USE_TENSORRT, to save some binary size when not using TensorRT.